services/src/main/scala/cromwell/services/metadata/impl/WriteMetadataActor.scala

```diff
@@ -29,8 +30,8 @@ final case class WriteMetadataActor(batchRate: Int, flushRate: FiniteDuration)
   when(WaitingToWrite) {
     case Event(PutMetadataAction(events), curData) =>
       curData.addEvents(events) match {
-        case data@HasEvents(e) if e.length > batchRate => goto(WritingToDb) using data
-        case e => stay using e
+        case newData: HasEvents if newData.length > batchRate => goto(WritingToDb) using newData
```

Comment:
is the switcharoo with the `@` and `length` due to the tests? If not i'm not getting why you did this.